src/twisted/logger/_observer.py

```diff
@@ -51,7 +51,7 @@ def removeObserver(self, observer: ILogObserver) -> None:
         """
         try:
             self._observers.remove(observer)
-        except ValueError:
+        except (KeyError, ValueError):
```

Comment:
Do we still get `ValueError` here ?

I think that `list.remove` was raising ValueError but `set.remove()` should only raise KeyError

 ```suggestion
         except KeyError:
```

Reply:
You are right, I updated